### PR TITLE
[+] Search - Display highlighted matches on table view when searching

### PR DESCRIPTION
**Pull request by root-io**
## Pull Request checklist:

- [x] Write an explicit title for the Pull Request
- [x] Write changes made in the CHANGELOG.md
- [x] Test manually the implemented changes
- [x] Review my own code (indentation, syntax, style, simplicity, readability)
- [x] Wonder if you can improve the existing code

### CODE REVIEW

**Comment by root-io** [comment]
[(13) Search - Users want to see highlighted matches on table view when searching](https://trello.com/c/h2GOQgdC/2168-13-search-users-want-to-see-highlighted-matches-on-table-view-when-searching)

File: utils/search.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+exports.getSearchMatchFields = function (searchString, records) {
+  records = records.map((record) => record.get({ plain: true }));
+  const matchFields = {};
+  records.forEach((record, index) => {
+    Object.entries(record).forEach(([key, value]) => {
```
Comment (arnaudbesnier) [inline utils/search.js]: `key` => `attributeName`

File: utils/search.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      if (value) {
+        value = value.toString();
+        const match = value.match(new RegExp(searchString, 'i'));
+        if (match) {
+          if (!matchFields[index]) {
+            matchFields[index] = { 'id': record.id };
+            matchFields[index]['search'] = [];
```
Comment (arnaudbesnier) [inline utils/search.js]:
```javascript
matchFields[index] = {
  id: record.id,
  search: [],
};
```

File: routes/resources.js [hunk 3]
```diff
@@ -19,13 +21,25 @@ module.exports = function (app, model, Implementation, integrator, opts) {
...
         var records = results[1];
 
+        var decorators = null;
+
+        if (params.search) {
+          var searchDecorator = SearchUtil.getSearchMatchFields(params.search, records);
+          if (!_.isEmpty(searchDecorator)) {
+            decorators = searchDecorator;
+          }
+        }
```
Comment (arnaudbesnier) [inline routes/resources.js]:
```javascript
var meta = {
  count: count,
};

if (params.search) {
  var decoratorsSearch = SearchUtil.getSearchMatchFields(params.search, records);
  if (!_.isEmpty(decoratorsSearch)) {
    meta.decorators = decoratorsSearch;
  }
}
```

File: routes/resources.js [hunk 4]
```diff
@@ -19,13 +21,25 @@ module.exports = function (app, model, Implementation, integrator, opts) {
...
           Implementation,
           model,
           records,
           integrator,
           opts,
-          { count: count },
+          {
+            count: count,
+            decorators: decorators,
+          },
```
Comment (arnaudbesnier) [inline routes/resources.js]: then use `meta` here

File: utils/search.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+'use strict';
```
Comment (arnaudbesnier) [inline utils/search.js]: I would call this file `utils/records-decorator.js` or `service/records-decorator.js`

File: utils/search.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+exports.getSearchMatchFields = function (searchString, records) {
```
Comment (arnaudbesnier) [inline utils/search.js]: and call this `decorateForSearch(records, searchValue)`

**Pull request closed without merging**